com.unity.multiplayer.mlapi/Runtime/Interest/RadiusInterestStorage.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI.Connection;
+using UnityEngine;
+
+namespace MLAPI.Interest
+{
+    [CreateAssetMenu(fileName = "RadiusStorage", menuName = "Interest/Storage/Radius", order = 1)]
+    [Serializable]
+    public class RadiusInterestStorage : BasicInterestStorage
+    {
+        public float Radius = 0.0f;
+        public override void Query(in NetworkClient client, HashSet<NetworkObject> results)
```

Comment:
I see the usage pattern so far but I also wonder how one could cache/invalidate query results to optimize queries?
maybe values/props my query is looking at are not changing too much and I don't want to spend CPU cycles?
did we start to think about bigger-picture/bigger-scope algorithm optimizations?

Reply:
I think there are 2 levels here, and I think I only cover one of them.

1) on a per-node basis, you can certainly implement a caching scheme.  The interface only requires you to accept a client and return results.  You can have your own internal dirty state and cached results and just return those (and avoid the compute).  Now I do want to explore other more efficient ways to return results than merging HashSet's.  That way, if a node has its own cache result it wants to quickly return it can do so.  Maybe a list of results, and then it just is appending a reference.  Or better yet, maybe a map where the key is the node itself, and then if the result hasn't changed you don't even touch it.

2) on a per-all-node basis, one could say "I can somehow tell that client 123 has the same answer as last time because none of the nodes are dirty, so I can just return the same result".  And/or, this could be done for all clients I suppose.  This type I'm less motivated to explore just because it seems almost as good to have caching in the nodes above, and the work to know if we can save the traversal seems a bit daunting.  What do you think?